tests/client/amo/components/TestLandingPage.js

```diff
@@ -101,9 +101,30 @@ describe('<LandingPage />', () => {
     );
   });
 
-  it('throws if add-on type is not supported', () => {
+  it('renders not found if add-on type is not supported', () => {
+    const root = render({ addonType: 'XUL' });
+    assert.include(root.textContent, 'Page not found');
+  });
+
+  it('renders not found if add-on type is not supported', () => {
+    assert.throws(() => {
+      // Stubbing in a fake i18n here will cause contentForType to throw.
+      // This is easier/lazier than doing dependency injection for that method.
+      render({ addonType: 'doesntmatter', i18n: null });
```

Comment:
You might be able to delete this test if you end up moving the visibleAddonType logic around as I suggested but I'm not sure